Project/Gem/Source/Scripting/FoilWrapperController.cpp

```diff
@@ -0,0 +1,166 @@
+
+        // Connect the trigger handlers if not already connected, it is circumventing the issue GH-16188, the
+        // RigidbodyNotificationBus should be used instead.
+        if (!m_onTriggerEnterHandler.IsConnected())
+        {
+            AZ_Assert(m_configuration.m_foilWrapperEntityId.IsValid(), "Invalid foil wrapper entity id.");
+            if (m_configuration.m_foilWrapperEntityId.IsValid())
+            {
+                AZStd::pair<AzPhysics::SceneHandle, AzPhysics::SimulatedBodyHandle> foundBody =
+                    physicsSystem->FindAttachedBodyHandleFromEntityId(m_configuration.m_collisionTrigger);
+                AZ_Warning(
+                    "FoilWrapper", foundBody.first != AzPhysics::InvalidSceneHandle, "No body found for m_collisionTrigger.");
+                if (foundBody.first != AzPhysics::InvalidSceneHandle)
+                {
+                    AzPhysics::SimulatedBodyEvents::RegisterOnTriggerEnterHandler(
+                        foundBody.first, foundBody.second, m_onTriggerEnterHandler);
+                }
+
+            }
+        }
+        if (m_state == FoilWrapperState::Idle)
+        {
+            m_timer = 0;
+        }
+        if (m_state == FoilWrapperState::Starting)
+        {
+            m_timer += delta;
+            if (m_timer> 10.0)
+            {
+                AZ_Printf("FoilWrapper", "Warpping starterd");
+                EMotionFX::Integration::SimpleMotionComponentRequestBus::Event(
+                    m_configuration.m_foilWrapperEntityId, &EMotionFX::Integration::SimpleMotionComponentRequestBus::Events::PlayMotion);
+                m_state = FoilWrapperState::Wrapping;
+                m_timer = 0;
+            }
+        }
+        if (m_state == FoilWrapperState::Wrapping)
+        {
+            m_timer += delta;
+            if (m_timer > 10.0)
+            {
+                AZ::Transform transform = AZ::Transform::CreateIdentity();
+                AZ::TransformBus::EventResult(transform, m_configuration.m_collisionTrigger, &AZ::TransformBus::Events::GetWorldTM);
+                for (auto& entityId : m_collidingEntities)
+                {
+                    AZ::Entity* entity{};
+                    AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationRequests::FindEntity, entityId);
+                    if (entity)
+                    {
+                        const auto& entityName = entity->GetName();
+                        AZ_Printf("FoilWrapper", "Despawning box with entity with name %s", entityName.c_str());
+                        ScriptSpawnSystemRequestBus::Broadcast(&ScriptSpawnSystemRequestBus::Events::DespawnBox,entityName);
+
+                    }
+                }
+                ScriptSpawnSystemRequestBus::Broadcast(&ScriptSpawnSystemRequestBus::Events::SpawnAsset, m_configuration.m_spawnablePayloadFoiled,transform, "foo");
```

Comment:
The name of the spawned foiled pallet should be changed. It can be similar to the box name (containing the time of spawning).